src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialMigrationTest.java

```diff
@@ -0,0 +1,115 @@
+import org.junit.rules.TemporaryFolder;
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+
+
+/**
+ *
+ * @author arroyc
+ */
+public class CredentialMigrationTest {
+
+    public static File input = new File("src/test/resources/com/microsoftopentechnologies/helper/correctFormatOldConfig.xml");
+    private Jenkins jenkinsInstance;
+    
+    @ClassRule
+    public static TemporaryFolder testFolder = new TemporaryFolder();
+    @ClassRule 
+    public static JenkinsRule j = new JenkinsRule();
+
+    @Before
+    public void setUp() {
+        jenkinsInstance = Jenkins.getInstance();
+        
+    }
+    /**
+     * Test of upgradeStorageConfig method, of class CredentialMigration.
+     */
+    @Test
+    public void testUpgradeStorageConfig() throws Exception {
+        System.out.println("upgradeStorageConfig");
+        
+        String storageAccount = "abcdef";
+        String storageAccountKey = "12345";
+        String storageBlobURL = "http://test1/";
+        File tempOldConfigFile = testFolder.newFile(Constants.LEGACY_STORAGE_CONFIG_FILE);
+        FileUtils.copyFile(input, tempOldConfigFile);
+        FileUtils.copyFileToDirectory(tempOldConfigFile, jenkinsInstance.root);
+        System.out.println(testFolder.getRoot().getAbsolutePath());
+
+        CredentialMigration.upgradeStorageConfig();
+        
+        CredentialsStore s = CredentialsProvider.lookupStores(jenkinsInstance).iterator().next();
+
+        assertEquals(2, s.getCredentials(Domain.global()).size());
+                
+        AzureCredentials.StorageAccountCredential u = new AzureCredentials.StorageAccountCredential(storageAccount, storageAccountKey, storageBlobURL);
+        AzureCredentials storageCred = CredentialsMatchers.firstOrNull(CredentialsProvider.lookupCredentials(AzureCredentials.class, jenkinsInstance, ACL.SYSTEM, Collections.<DomainRequirement>emptyList()),
+                CredentialsMatchers.withId(u.getId()));
+        
+        //System.out.println("StorageAccountName: "+cred.getStorageAccountName()+", StorageAccountKey: "+cred.getStorageKey()+", BlobURL: "+cred.getBlobEndpointURL());
+
+        assertEquals(u.getStorageAccountName(), storageCred.getStorageAccountName());
+        assertEquals(u.getEndpointURL(), storageCred.getBlobEndpointURL());
+        assertEquals(u.getSecureKey().getEncryptedValue(), storageCred.getStorageKey());
+        
+    }
+
+    /**
+     * Test of getOldStorageConfig method, of class CredentialMigration.
+     */
+    @Test
+    public void testGetOldStorageConfig() throws Exception {
+        System.out.println("getOldStorageConfig");
+        //File input = new File("src/test/resources/com/microsoftopentechnologies/helper/correctFormatOldConfig.xml");
```

Comment:
the correctFormatOldConfig file will get shipped with the hpi. I don't think we want that. I think it's better to hardcode the content in one of the test files and just write it when needed